urllib3/util/selectors.py

```diff
@@ -112,10 +117,9 @@ def _syscall_wrapper(func, recalc_timeout, *args, **kwargs):
                     if expires is not None:
                         current_time = monotonic()
                         if current_time > expires:
-                            raise OSError(errno=errno.ETIMEDOUT)
+                            raise OSError(errno.ETIMEDOUT, "Connection timed out")
                         if recalc_timeout:
-                            if "timeout" in kwargs:
-                                kwargs["timeout"] = expires - current_time
+                            kwargs["timeout"] = expires - current_time
```

Comment:
I'm trying to write a test locally right now to reproduce this issue, I'll reply back with my results.